catalyst-gateway/bin/src/service/api/config/mod.rs

```diff
@@ -0,0 +1,166 @@
+
+    /// Get the frontend JSON schema.
+    #[oai(
+        path = "/draft/config/frontend/schema",
+        method = "get",
+        operation_id = "get_config_frontend_schema"
+    )]
+    #[allow(clippy::unused_async)]
+    async fn get_frontend_schema(
+        &self, #[oai(name = "IP")] ip_query: Query<Option<String>>,
+    ) -> Responses {
+        match ip_query.0 {
+            Some(ip) => {
+                match IpAddr::from_str(&ip) {
+                    Ok(parsed_ip) => {
+                        Responses::Ok(Json(ConfigKey::FrontendForIp(parsed_ip).schema().clone()))
+                    },
+                    Err(err) => {
+                        Responses::BadRequest(Json(json!(format!("Invalid IP address: {err}"))))
+                    },
+                }
+            },
+            None => Responses::Ok(Json(ConfigKey::Frontend.schema().clone())),
+        }
+    }
+
+    /// Set the frontend configuration.
+    #[oai(
+        path = "/draft/config/frontend",
+        method = "put",
+        operation_id = "put_config_frontend"
+    )]
+    async fn put_frontend(
+        &self, #[oai(name = "IP")] ip_query: Query<Option<String>>, body: Json<Value>,
+    ) -> Responses {
+        let body_value = body.0;
+
+        match ip_query.0 {
+            Some(ip) => {
+                match IpAddr::from_str(&ip) {
+                    Ok(parsed_ip) => set(ConfigKey::FrontendForIp(parsed_ip), body_value).await,
+                    Err(err) => {
+                        Responses::BadRequest(Json(json!(format!("Invalid IP address: {err}"))))
+                    },
+                }
+            },
+            None => set(ConfigKey::Frontend, body_value).await,
+        }
+    }
+}
+
+/// Helper function to merge two JSON values.
+fn merge_configs(general: &Value, ip_specific: &Value) -> Value {
+    let mut merged = general.clone();
+
+    if let Some(ip_specific_obj) = ip_specific.as_object() {
+        if let Some(merged_obj) = merged.as_object_mut() {
+            for (key, value) in ip_specific_obj {
+                if let Some(existing_value) = merged_obj.get_mut(key) {
+                    *existing_value = value.clone();
+                } else {
+                    merged_obj.insert(key.clone(), value.clone());
+                }
+            }
+        }
+    }
+
+    merged
+}
+
+/// Helper function to handle set.
+async fn set(key: ConfigKey, value: Value) -> Responses {
+    match Config::set(key, value).await {
+        Ok(validate) => {
+            match validate {
+                BasicOutput::Valid(_) => {
+                    Responses::Ok(Json(json!("Configuration successfully set.")))
+                },
+                BasicOutput::Invalid(errors) => {
+                    let mut e = vec![];
+                    for error in errors {
+                        e.push(error.error_description().to_string());
+                    }
+                    Responses::BadRequest(Json(json!({"errors": e})))
```

Comment:
You should define an "Error" response type for 400, that is used for any 400 that can be returned, so that the structure is the same.  Otherwise the frontend has no way to know how to parse it.